### PR TITLE
[boschshc] Add support for motion detector illuminance sensor

### PR DESCRIPTION
**Pull request by david-pace**
- add channel and corresponding channel type
- add update description
- add state model
- implement service and handler
- add documentation
- add unit test

### CODE REVIEW

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/update/binding.xml [hunk 1]
```diff
@@ -12,4 +13,13 @@
 			</add-channel>
 		</instruction-set>
 	</thing-type>
+
+	<thing-type uid="boschshc:motion-detector">
+		<instruction-set targetVersion="1">
```
Comment (lolodomo) [inline bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/update/binding.xml:18]: You need to add the property `thingTypeVersion` in your thing type (XML file).

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java [hunk 2]
```diff
@@ -43,10 +48,16 @@ protected void initializeServices() throws BoschSHCException {
...
+        this.createService(IlluminanceService::new, this::updateChannels, List.of(CHANNEL_ILLUMINANCE), true);
     }
 
     private void updateChannels(LatestMotionServiceState state) {
         DateTimeType date = new DateTimeType(state.latestMotionDetected);
         updateState(CHANNEL_LATEST_MOTION, date);
     }
+
+    private void updateChannels(IlluminanceServiceState state) {
+        DecimalType illuminance = new DecimalType(state.illuminance);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java:60]: By curiosity, is it a percentage or not ?
If a percentage, you could have used `PercentType` and add a default pattern including the % in the XML definition.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java (hunk 2 above)
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java:60]: Thanks for asking 🙂 When I first tested the sensor, I thought the values were in the 0-100 range. But now I wanted to be sure and held a very bright light directly in front of the sensor. The resulting value: `874` 😜 So probably the range is 0 - 1000. I adapted the code and the documentation accordingly.

General question, not related to this PR: we have a few devices that report battery levels. We use system channels with type `system.battery-level` for that, but we don't use `PercentType` values but `DecimalType` values. Should this be corrected in another pull request?

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/motiondetector/MotionDetectorHandler.java:60]: > General question, not related to this PR: we have a few devices that report battery levels. We use system channels with type system.battery-level for that, but we don't use PercentType values but DecimalType values. Should this be corrected in another pull request?

You can use both I believe.

**Pull request merged**